core/wasm/src/cache.rs

```diff
@@ -0,0 +1,13 @@
+use tempfile::TempDir;
+use std::sync::Mutex;
+use std::path::{Path, PathBuf};
+
+lazy_static! {
+    static ref CACHE_TEMP_DIR: Mutex<TempDir> =
+        Mutex::new(TempDir::new()
```

Comment:
I do not think `lazy_static` calls destructor on the underlying object and so the temporary directory will not be cleaned up, polluting the disk.

Reply:
https://github.com/rust-lang-nursery/lazy-static.rs/issues/62

Reply:
We need some object with a regular lifetime (such that all `execute` calls are within that lifetime by our design), that creates and cleans the temporary directory. It would then store the path to that directory in variable that itself is wrapped in `lazy_static` and `Option`. The `execute` code then would read the path by unwrapping the `Option` just like it does now with `get_cached_path`.

The difference basically is that we need to store the path in `lazy_static` and not the actual handle to the directory. The handle should be attached to some object with a regular lifetime.

Reply:
Also consider using `std::sync::RwLock` instead of `Mutex`.

Reply:
Replaced on-disk cache with in-memory LRU cache. But it's still static

Reply:
PTAL

Reply:
LGTM